course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala

```diff
@@ -293,12 +309,25 @@ class ContentConsumptionActor @Inject() extends BaseEnrolmentActor {
         val courseId = request.get(JsonKey.COURSE_ID).asInstanceOf[String]
         val contentIds = request.getRequest.getOrDefault(JsonKey.CONTENT_IDS, new java.util.ArrayList[String]()).asInstanceOf[java.util.List[String]]
         val fields = request.getRequest.getOrDefault(JsonKey.FIELDS, new java.util.ArrayList[String](){{ add(JsonKey.PROGRESS) }}).asInstanceOf[java.util.List[String]]
-        val contentsConsumed = getContentsConsumption(userId, courseId, contentIds, batchId, request.getRequestContext)
+        //default fields added ..
+        if (fields!=null && fields.contains(progressDetails) && !defaultFields.contains(progressdetails)){
+            defaultFields.add(progressdetails)
+        } else if(fields!=null && !fields.contains(progressdetails)){
+            defaultFields.remove(progressdetails)
+        }
+
+        val contentsConsumed = getContentsConsumption(userId, courseId, contentIds, batchId, defaultFields, request.getRequestContext)
         val response = new Response
         if(CollectionUtils.isNotEmpty(contentsConsumed)) {
             val filteredContents = contentsConsumed.map(m => {
                 ProjectUtil.removeUnwantedFields(m, JsonKey.DATE_TIME, JsonKey.USER_ID, JsonKey.ADDED_BY, JsonKey.LAST_UPDATED_TIME)
                 m.put(JsonKey.COLLECTION_ID, m.getOrDefault(JsonKey.COURSE_ID, ""))
+                //added progress starts....
+                if(fields.contains(progressDetails)){
+                    m.put(progressDetails, mapper.readTree(m.get(progressdetails).asInstanceOf[String]))
+                    m.remove(progressdetails)
+                }
```

Comment:
@pritha-tarento - Same as the comment earlier. Generalize this code

Reply:
Resolved